samples/subsys/zbus/hello_world/src/main.c

```diff
@@ -0,0 +1,158 @@
+
+		 NULL,				       /* Validator */
+		 ZBUS_OBSERVERS(foo_lis, bar_sub),     /* observers */
+		 ZBUS_MSG_INIT(.x = 0, .y = 0, .z = 0) /* Initial value */
+);
+
+bool simple_chan_validator(void *msg, size_t msg_size)
+{
+	ARG_UNUSED(msg_size);
+
+	int *simple = msg;
+
+	if ((*simple >= 0) && (*simple < 10)) {
+		return true;
+	}
+
+	return false;
+}
+
+ZBUS_CHAN_DEFINE(simple_chan, /* Name */
+		 false,	      /* Read only */
+		 int,	      /* Message type */
+
+		 simple_chan_validator, /* Validator */
+		 ZBUS_OBSERVERS_EMPTY,	/* observers */
+		 0			/* Initial value is 0 */
+);
+
+void listener_callback_example(struct zbus_channel *chan)
+{
+	struct acc_msg *acc = zbus_chan_msg(chan);
+
+	LOG_INF("From listener -> Acc x=%d, y=%d, z=%d", acc->x, acc->y, acc->z);
+}
+
+ZBUS_LISTENER_DEFINE(foo_lis, listener_callback_example);
+
+ZBUS_SUBSCRIBER_DEFINE(bar_sub, 4);
+
+void subscriber_task(void)
+{
+	struct zbus_channel *chan = NULL;
+
+	while (!zbus_sub_wait(&bar_sub, &chan, K_FOREVER)) {
+		struct acc_msg acc = {0};
+
+		if (&acc_data_chan == chan) {
+			zbus_chan_read(&acc_data_chan, &acc, K_MSEC(500));
+
+			LOG_INF("From subscriber -> Acc x=%d, y=%d, z=%d", acc.x, acc.y, acc.z);
+		}
+	}
+}
+
+K_THREAD_DEFINE(subscriber_task_id, 512, subscriber_task, NULL, NULL, NULL, 3, 0, 0);
+
+int count;
+
+bool print_channel_data_iterator(struct zbus_channel *chan)
+{
+	LOG_INF("%d - Channel %s:", count, chan->name);
+	LOG_INF("      Message size: %d", chan->message_size);
+	++count;
+	LOG_INF("      Observers:");
+	for (struct zbus_observer **obs = chan->observers; *obs != NULL; ++obs) {
+		LOG_INF("      - %s", (*obs)->name);
+	}
+
+	return true;
+}
+
+bool print_observer_data_iterator(struct zbus_observer *obs)
+{
+	LOG_INF("%d - %s %s", count, ((obs->queue != NULL) ? "Subscriber" : "Listener"), obs->name);
+
+	++count;
+
+	return true;
+}
+
+void main(void)
+{
+	int err, value;
+	struct acc_msg acc1 = {.x = 1, .y = 1, .z = 1};
+	struct version_msg *v = zbus_chan_msg(&version_chan);
+
+	LOG_INF("Sensor sample started raw reading, version %u.%u-%u!", v->major, v->minor,
+		v->build);
+
+#if defined(CONFIG_ZBUS_STRUCTS_ITERABLE_ACCESS) && CONFIG_ZBUS_STRUCTS_ITERABLE_ACCESS == 1
+	count = 0;
+
+	LOG_INF("Channel list:");
+	zbus_iterate_over_channels(print_channel_data_iterator);
+
+	count = 0;
+
+	LOG_INF("Observers list:");
+	zbus_iterate_over_observers(print_observer_data_iterator);
+#endif
+	zbus_chan_pub(&acc_data_chan, &acc1, K_SECONDS(1));
+
+	k_msleep(1000);
+
+	acc1.x = 2;
+	acc1.y = 2;
+	acc1.z = 2;
+	zbus_chan_pub(&acc_data_chan, &(acc1), K_SECONDS(1));
+
+	value = 5;
+	err = zbus_chan_pub(&simple_chan, &value, K_MSEC(200));
+
+	LOG_INF("Pub a valid value to a channel with validator. Return %serror",
+		err == 0 ? "no " : "");
+
+	value = 15;
+	err = zbus_chan_pub(&simple_chan, &value, K_MSEC(200));
+
+	LOG_INF("Pub an invalid value to a channel with validator. Return %serror",
+		err == -ENOMSG ? "" : "no ");
```

Comment:
I find this pattern of capturing the error code and reducing it to a "yes/no" a bit awkard. Imagine this failing with another error code and having to debug it, you'd end up having to change the sample to print the actual code. I would rather just log the return value as it is, (like instead of `Return %serror` something like `ret = %d` and the code). Easier to spot than a "no" in the middle of the log message as well. Same for the one above.

Reply:
Fixed